packages/js/product-editor/src/blocks/generic/collapsible/edit.tsx

```diff
@@ -30,3 +35,63 @@ export function Edit( {
 		</div>
 	);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const areAllBlocksInvisible = ( blocks: BlockInstance[], context: any ) => {
+	return blocks.every( ( block ) => {
+		if (
+			! block.attributes?._templateBlockHideConditions ||
+			! Array.isArray( block.attributes?._templateBlockHideConditions )
+		) {
+			return false;
+		}
+		return block.attributes._templateBlockHideConditions.some(
+			( condition ) => evaluate( condition.expression, context )
+		);
+	} );
+};
+
+const maybeHideInventoryAdvancedCollapsible = createHigherOrderComponent<
+	Record< string, unknown >
+>( ( BlockEdit ) => {
+	return ( props ) => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const evalContext = useEvaluationContext( props.context as any );
+
+		if (
+			( props?.attributes as Record< string, string > )
+				?._templateBlockId !== 'product-inventory-advanced'
+		) {
+			return <BlockEdit { ...props } />;
+		}
+
+		// get the inventory section block instance
+		const advancedCollapsibleBlock = select( 'core/block-editor' ).getBlock(
+			props?.clientId as string
+		);
+
+		// No inner blocks, so we can render the default block edit.
+		if ( ! advancedCollapsibleBlock?.innerBlocks?.length ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		const advancedSectionBlock = advancedCollapsibleBlock?.innerBlocks[ 0 ];
+
+		if (
+			areAllBlocksInvisible(
+				advancedSectionBlock?.innerBlocks,
+				evalContext.getEvaluationContext( select )
+			)
+		) {
+			return null;
+		}
+
+		return <BlockEdit { ...props } />;
+	};
+}, 'maybeHideInventoryAdvancedCollapsible' );
+
+addFilter(
+	'editor.BlockEdit',
+	'woocommerce/handle-hide-inventory-advanced-collapsible',
+	maybeHideInventoryAdvancedCollapsible
```

Comment:
Is there a specific reason you decided to go the `filter` route on this one?
It does look like we would run into a similar scenario for the section field if you try this for the images section:
```
function remove_limit_purchase_block($block)
{
    $block->remove();
}

add_action('woocommerce_block_template_area_product-form_after_add_block_product-images', 'remove_limit_purchase_block');
```

Maybe it would be worth moving this to a more generic location and adding the `section` field also as an exception.